cluster-autoscaler/core/scale_down_test.go

```diff
@@ -794,6 +806,47 @@ func TestDrainNodeWithRetries(t *testing.T) {
 	assert.Equal(t, p3.Name, deleted[2])
 }
 
+func TestDrainNodeDsEvictionFailure(t *testing.T) {
+	fakeClient := &fake.Clientset{}
+
+	p1 := BuildTestPod("p1", 100, 0)
+	p2 := BuildTestPod("p2", 300, 0)
+	d1 := BuildTestPod("d1", 150, 0)
+	d2 := BuildTestPod("d2", 250, 0)
+	n1 := BuildTestNode("n1", 1000, 1000)
+	e1 := fmt.Errorf("eviction_error: d1")
+	e2 := fmt.Errorf("eviction_error: d2")
+
+	fakeClient.Fake.AddReactor("get", "pods", func(action core.Action) (bool, runtime.Object, error) {
+		return true, nil, errors.NewNotFound(apiv1.Resource("pod"), "whatever")
+	})
+	fakeClient.Fake.AddReactor("create", "pods", func(action core.Action) (bool, runtime.Object, error) {
+		createAction := action.(core.CreateAction)
+		if createAction == nil {
+			return false, nil, nil
+		}
+		eviction := createAction.GetObject().(*policyv1.Eviction)
+		if eviction == nil {
+			return false, nil, nil
+		}
+		if eviction.Name == "d1" {
+			return true, nil, e1
+		}
+		if eviction.Name == "d2" {
+			return true, nil, e2
+		}
+		return true, nil, nil
+	})
+	evictionResults, err := drainNode(n1, []*apiv1.Pod{p1, p2}, []*apiv1.Pod{d1, d2}, fakeClient, kube_util.CreateEventRecorder(fakeClient), 20, 0*time.Second, 0*time.Second, PodEvictionHeadroom)
+	assert.NoError(t, err)
+	assert.Equal(t, 2, len(evictionResults))
+	assert.Equal(t, *p1, *evictionResults["p1"].Pod)
+	assert.Equal(t, *p2, *evictionResults["p2"].Pod)
+	assert.NoError(t, evictionResults["p1"].Err)
```

Comment:
Why are you checking Err, TimedOut and WasEvictionSuccessful for 1 pod only? If the assertions hold true for both, please assert both because otherwise the reader is wondering what's different between the 2 pods.